api/src/main/java/com/google/sps/Notifications.java

```diff
@@ -0,0 +1,141 @@
+  private static final String CREDENTIALS_FILE_PATH = "./credentials.json";
+
+  /**
+   * Creates an authorized Credential object.
+   *
+   * @param HTTP_TRANSPORT The network HTTP Transport.
+   * @return An authorized Credential object.
+   * @throws IOException If the credentials.json file cannot be found.
+   */
+  private static HttpRequestInitializer getCredentials(NetHttpTransport HTTP_TRANSPORT) throws IOException {
+    // Load client secrets.
+    Path currentRelativePath = Paths.get("");
+    String s = currentRelativePath.toAbsolutePath().toString();
+    System.out.println("Current relative path is: " + s);
+
+    File file = ResourceUtils.getFile(CREDENTIALS_FILE_PATH);
+    if (file == null) {
+      throw new FileNotFoundException("Resource not found: " + CREDENTIALS_FILE_PATH);
+    }
+    InputStream in = new FileInputStream(file);
+    GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(JSON_FACTORY, new InputStreamReader(in));
+
+    // Build flow and trigger user authorization request.
+    GoogleAuthorizationCodeFlow flow = new GoogleAuthorizationCodeFlow.Builder(HTTP_TRANSPORT, JSON_FACTORY,
+        clientSecrets, SCOPES).setDataStoreFactory(new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+            .setAccessType("offline").build();
+    LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(8888).build();
+
+    return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+  }
+
+  private static MimeMessage createEmail(String to, String from, String subject, String bodyText)
+      throws MessagingException {
+    Properties props = new Properties();
+    Session session = Session.getDefaultInstance(props, null);
+
+    MimeMessage email = new MimeMessage(session);
+
+    email.setFrom(new InternetAddress(from));
+    email.addRecipient(javax.mail.Message.RecipientType.TO, new InternetAddress(to));
+    email.setSubject(subject);
+    email.setText(bodyText);
+    return email;
+  }
+
+  private static Message createMessageWithEmail(MimeMessage emailContent) throws MessagingException, IOException {
+    ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+    emailContent.writeTo(buffer);
+    byte[] bytes = buffer.toByteArray();
+    String encodedEmail = Base64.encodeBase64URLSafeString(bytes);
+    Message message = new Message();
+    message.setRaw(encodedEmail);
+    return message;
+  }
+
+  private static Message sendMessage(Gmail service, String userId, MimeMessage emailContent)
+      throws MessagingException, IOException {
+    Message message = createMessageWithEmail(emailContent);
+    message = service.users().messages().send(userId, message).execute();
+
+    System.out.println("Message id: " + message.getId());
+    System.out.println(message.toPrettyString());
+        return message;
+    }
+
+    public static void sendNotification() throws IOException, GeneralSecurityException {
```

Comment:
Nit: Extra spaces?

I think there should be some auto indentation tool which you could use.

Reply:
fixed, i'll try to find one